pkg/api/handlers.go

```diff
@@ -297,15 +298,18 @@ func NewQuerierHandler(
 	cardinalityQueryStats := usagestats.NewRequestsMiddleware("querier_cardinality_query_requests")
 	formattingQueryStats := usagestats.NewRequestsMiddleware("querier_formatting_requests")
 
+	// Add memory consumption tracker in middleware for endpoints like /read and /series that need memory consumption tracker to be available in context.
```

Comment:
Could you please expand the comment here to explain why we need a memory consumption tracker, and why we can't do this somewhere else?